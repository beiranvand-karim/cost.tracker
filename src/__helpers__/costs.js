 const costs = [
   {
      cost: 1000,
      comment: 'this is a comment',
      category: 'CLOTHING'
   },
   {
      cost: 1500,
      comment: 'this is a comment',
      category: 'RENT'
   },
   {
      cost: 2000,
      comment: 'this is a comment',
      category: 'FOOD'
   }
];

export default costs;
