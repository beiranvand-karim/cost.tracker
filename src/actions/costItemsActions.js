export const ADD_FILTERED_COST_ITEMS = "ADD_FILTERED_COST_ITEMS";
export const ADD_COST_ITEM = "ADD_COST_ITEM";


export const addFilteredCostItemsAction = data => ({
  type: ADD_FILTERED_COST_ITEMS,
  payload: data
});

export const addCostItemAction = data => ({
  type: ADD_COST_ITEM,
  payload: data
});
