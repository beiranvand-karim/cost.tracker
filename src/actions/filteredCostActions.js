export const ADD_COST = 'ADD_COST';
export const ADD_FILTERED_COSTS = 'ADD_FILTERED_COSTS';

export const addCostAction = (data) => {
   return {
      type: ADD_COST,
      payload: data
   }
};

export const addFilteredActions = (data) => {
   return {
      type: ADD_FILTERED_COSTS,
      payload: data
   }
};
