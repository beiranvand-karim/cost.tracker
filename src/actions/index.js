export {
  addCostItemAction,
  addFilteredCostItemsAction
} from "./costItemsActions";
export { addSourceCostItemAction } from "./sourceCostItemsActions";
