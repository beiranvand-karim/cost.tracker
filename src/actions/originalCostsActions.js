export const ADD_ORIGINAL_COST = 'ADD_ORIGINAL_COST';

export const addOriginalCostAction = (data) => {
   return {
      type: ADD_ORIGINAL_COST,
      payload: data
   }
};
