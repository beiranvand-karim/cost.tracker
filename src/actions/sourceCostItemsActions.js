export const ADD_SOURCE_COST_ITEM = "ADD_SOURCE_COST_ITEM";

export const addSourceCostItemAction = data => ({
  type: ADD_SOURCE_COST_ITEM,
  payload: data
});
