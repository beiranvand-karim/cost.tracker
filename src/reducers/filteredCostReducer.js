import {ADD_COST, ADD_FILTERED_COSTS} from '../actions/filteredCostActions';

const defaultState = {
  costs: []
};

export default function filteredCostReducer(state = defaultState, action) {
   switch (action.type) {
      case ADD_COST:
         const {costs} = state;
         costs.unshift(action.payload);
         return {
            ...state,
            costs
         };
      case ADD_FILTERED_COSTS:
         return {
            ...state,
            costs: action.payload
         };
      default:
         return state;
   }
}
