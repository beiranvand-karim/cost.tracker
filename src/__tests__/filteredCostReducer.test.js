/* global expect, describe, it, beforeEach */
import { ADD_FILTERED_COSTS, ADD_COST} from '../actions/filteredCostActions'
import filteredCostReducer from '../reducers/filteredCostReducer'
import costs from "../__helpers__/costs";
import DUMMY_ACTION from "../constants/dummyAction";

describe('filteredCostReducer', () => {

   it('should matches the initial state', () => {
      const action = { type: DUMMY_ACTION };
      expect(filteredCostReducer(undefined, action)).toMatchSnapshot()
   });

   it('should matches the add filtered costs state', () => {
      const action = {
         type: ADD_FILTERED_COSTS,
         payload: costs
      };
      expect(filteredCostReducer(undefined, action)).toMatchSnapshot()
   });

   it('should matches the add cost state', () => {
      const action = {
         type: ADD_COST,
         payload: costs[0]
      };
      expect(filteredCostReducer(undefined, action)).toMatchSnapshot()
   });

});
