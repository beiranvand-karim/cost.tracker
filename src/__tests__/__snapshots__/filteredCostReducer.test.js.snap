// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filteredCostReducer should matches the add cost state 1`] = `
Object {
  "costs": Array [
    Object {
      "category": "CLOTHING",
      "comment": "this is a comment",
      "cost": 1000,
    },
  ],
}
`;

exports[`filteredCostReducer should matches the add filtered costs state 1`] = `
Object {
  "costs": Array [
    Object {
      "category": "CLOTHING",
      "comment": "this is a comment",
      "cost": 1000,
    },
    Object {
      "category": "RENT",
      "comment": "this is a comment",
      "cost": 1500,
    },
    Object {
      "category": "FOOD",
      "comment": "this is a comment",
      "cost": 2000,
    },
  ],
}
`;

exports[`filteredCostReducer should matches the initial state 1`] = `
Object {
  "costs": Array [],
}
`;
